lbry/wallet/orchstr8/node.py

```diff
@@ -429,3 +445,300 @@ def decode_raw_transaction(self, tx):
+            }
+        if "channel" == key:
+            kwargs["channel_id"] = {
+                "invert": False,
+                "value": kwargs["channel"]
+            }
+            del kwargs["channel"]
+        if "not_channel_id" == key:
+            kwargs["channel_id"] = {
+                "invert": True,
+                "value": kwargs["not_channel_id"]
+            }
+            del kwargs["not_channel_id"]
+        if "channel_ids" == key:
+            kwargs["channel_ids"] = {
+                "invert": False,
+                "value": kwargs["channel_ids"]
+            }
+        if "not_channel_ids" == key:
+            kwargs["channel_ids"] = {
+                "invert": True,
+                "value": kwargs["not_channel_ids"]
+            }
+            del kwargs["not_channel_ids"]
+
+
+        if key in MY_RANGE_FIELDS and isinstance(value, str) and value[0] in ops:
+            operator_length = 2 if value[:2] in ops else 1
+            operator, value = value[:operator_length], value[operator_length:]
+
+            op = 0
+            if operator == '=':
+                op = 0
+            if operator == '<=' or operator == 'lte':
+                op = 1
+            if operator == '>=' or operator == 'gte':
+                op = 2
+            if operator == '<' or operator == 'lt':
+                op = 3
+            if operator == '>' or operator == 'gt':
+                op = 4
+            kwargs[key] = {"op": op, "value": [str(value)]}
+        elif key in MY_RANGE_FIELDS:
+            kwargs[key] = {"op": 0, "value": [str(value)]}
+
+        if 'fee_amount' == key:
+            value = kwargs['fee_amount']
+            value.update({"value": [str(Decimal(value['value'][0]) * 1000)]})
+            kwargs['fee_amount'] = value
+        if 'stream_types' == key:
+            kwargs['stream_type'] = kwargs.pop('stream_types')
+        if 'media_types' == key:
+            kwargs['media_type'] = kwargs.pop('media_types')
+    return kwargs
+
+
+class HubNode:
+
+    def __init__(self, url, daemon, cli):
+        self.debug = True
+
+        self.latest_release_url = url
+        self.project_dir = os.path.dirname(os.path.dirname(__file__))
+        self.bin_dir = os.path.join(self.project_dir, 'bin')
+        self.daemon_bin = os.path.join(self.bin_dir, daemon)
+        self.cli_bin = os.path.join(self.bin_dir, daemon)
+        self.log = log.getChild('hub')
+        self.data_path = None
+        self.protocol = None
+        self.transport = None
+        self.block_expected = 0
+        self.hostname = 'localhost'
+        # self.peerport = 9246 + 13  # avoid conflict with default peer port
+        self.rpcport = 50051 # avoid conflict with default rpc port
+        self.rpcuser = 'rpcuser'
+        self.rpcpassword = 'rpcpassword'
+        self.stopped = False
+        self.restart_ready = asyncio.Event()
+        self.restart_ready.set()
+        self.running = asyncio.Event()
```

Comment:
This class needs some cleanup, still lots of remnants from when it was copied from `BlockchainNode`.